apps/gdalalg_vector_pipeline.cpp

```diff
@@ -548,6 +548,18 @@ bool GDALVectorPipelineAlgorithm::ParseCommandLineArguments(
         inputArg->GetType() == GAAT_DATASET_LIST &&
         inputArg->Get<std::vector<GDALArgDatasetValue>>().size() == 1)
     {
+        if (inputArg->GetType() == GAAT_DATASET_LIST)
```

Comment:
I don't know anything about the code/problem, but doesn't line 548 guarantee that this condition is always true?